dapp-oeth/src/components/buySell/SwapCurrencyPill.js

```diff
@@ -0,0 +1,881 @@
+          top: 0;
+          left: 0;
+          margin: 0 !important;
+          z-index: 9999;
+          display: flex;
+          flex-direction: column;
+          height: 100vh;
+          width: 100vw;
+          align-items: center;
+          justify-content: center;
+        }
+
+        --bg-opacity: 0.9;
+
+        .token-selection-modal .content-backdrop {
+          position: absolute;
+          top: 0;
+          left: 0;
+          display: flex;
+          flex-direction: column;
+          height: 100%;
+          width: 100%;
+          background-color: rgb(16 17 19 / var(--bg-opacity));
+        }
+
+        .token-selection-modal .content-container {
+          display: flex;
+          flex-direction: column;
+          margin: 0 auto;
+          max-height: 60vh;
+          width: 90vw;
+          max-width: 600px;
+          z-index: 2;
+          background-color: #1e1f25;
+          border-radius: 12px;
+          padding: 16px;
+          overflow: auto;
+          color: #fafbfb;
+        }
+
+        .token-selection-modal .coin-view {
+          display: flex;
+          flex-direction: row;
+          align-items: center;
+        }
+
+        .token-selection-modal .coin-breakdown {
+          display: flex;
+          flex-direction: column;
+          margin-left: 12px;
+        }
+
+        .coin-breakdown .name {
+          display: flex;
+          flex-direction: column;
+          font-size: 16px;
+          font-weight: 500;
+          color: #fafbfb;
+          text-align: left;
+        }
+
+        .token-selection-modal .coin-breakdown .coin {
+          display: flex;
+          flex-direction: column;
+          color: #828699;
+          font-size: 12px;
+          font-weight: 400;
+          text-align: left;
+        }
+
+        .token-selection-modal .coin .usd {
+          color: #828699;
+        }
+
+        .token-selection-modal .balances {
+          display: flex;
+          flex-direction: column;
+          color: #fafbfb;
+          font-size: 16px;
+          font-weight: 500;
+          text-align: right;
+        }
+
+        .balances .usd {
+          color: #828699;
+          font-size: 12px;
+          font-weight: 400;
+        }
+
+        .selectable {
+          border: none;
+          padding: 8px 6px;
+          background-color: transparent;
+          border-radius: 8px;
+        }
+
+        .selectable:hover {
+          background-color: #141519;
+        }
+      `}</style>
+    </>
+  )
+}
+
+const CoinSelect = ({
+  selected,
+  onChange,
+  options = [],
+  conversion,
+  coinBalances,
+}) => {
+  const [open, setOpen] = useState(false)
+
+  if (options.length === 0) {
+    return (
+      <>
+        <div
+          className={`coin-select d-flex align-items-center justify-content-start`}
+        >
+          <CoinImage coin={selected} />
+          <span className="coin mr-auto">
+            {selected === 'mix'
+              ? 'Mix LSDs'
```

Comment:
The contract doesn't always return LSDs when redeeming right? It can also return WETH